sw/device/lib/base/macros.h

```diff
@@ -98,4 +116,10 @@
  */
 #define OT_ATTR_WEAK __attribute__((weak))
 
+/**
+ * Attribute for weak alias that can be overridden, e.g., Mock overrides in
+ * DIFs.
+ */
+#define OT_ATTR_ALIAS(name) __attribute__((weak, alias(name)))
```

Comment:
I don't think you meant to include `weak` here right?

Reply:
No, I mean `weak` and `alias`.  I want the unit test to provide its own version for mocks.  I can rename the macro as `OT_ATTR_WEAK_ALIAS` if you prefer (IMHO, a weak alias is the only thing that really makes sense to me, but that opinion is kind-of driven by my use-case).

The documentation for `alias` doesn't lead me to believe that an alias is automatically weak.
https://gcc.gnu.org/onlinedocs/gcc-4.3.5/gcc/Function-Attributes.html

Reply:
No, what I meant is why you can't write
```c
OT_ATTR_WEAK
OT_ATTR_ALIAS(...)
void whatever();
```

AIUI attributes always combine together, and this is just a convenience syntax.

Reply:
Done.